src/locutus/model/terminology.py

```diff
@@ -572,6 +572,76 @@ def remove_pref(self, code=None):
 
         return message
 
+    def get_preferred_terminology(self):
+        """
+        Retrieves all references from the 'preferred_terminology' sub-collection
+
+        Returns:
+            dict: An array of `Terminology` references under the "references" key
+        
+        Example:
+        {
+            "reference": [
+                "/Terminology/tm--example1",
+                "/Terminology/tm--example3"]
+        ]    
+            
+        """
+        try:
+            doc_ref = persistence().collection(self.resource_type).document(self.id) \
+                .collection("preferred_terminology").document("self")
+
+            doc_snapshot = doc_ref.get()
+            if doc_snapshot.exists:
+                return doc_snapshot.to_dict()
+            else:
+                # If refs don't exist return an empty list
+                return {"reference": []}
+
+        except Exception as e:
+            print(f"An error occurred while retrieving preferred terminology: {e}")
+            raise
+
+
+    def add_preferred_terminology(self, preferred_terminology):
+        """
+        Creates or adds to a document in the 'preferred_terminology' sub-collection
+
+        Args:
+            preferred_terminology (dict): A dictionary representing the preferred terminology to be added.
+
+        JSON body example:
+        {
+            "preferred_terminology": "tm--example1"
+        }
+        """
+        try:
+            # Reference to the sub-collection document named "self"
+            doc_ref = persistence().collection(self.resource_type).document(self.id) \
+                .collection("preferred_terminology").document("self")
+
+            # Retrieve the current data if it exists
+            doc_snapshot = doc_ref.get()
+            if doc_snapshot.exists:
+                data = doc_snapshot.to_dict()
+                references = data.get("references", {})
+            else:
+                references = {}
+
+            # Create a new reference(unique key required - python)
+            key = f"reference_{len(references) + 1}"
+            new_pref = {key: f"/Terminology/{preferred_terminology}"}

```

Comment:
If these are actual reference strings, that "/" at the beginning doesn't need to be there.